src/emulator/modules/SceLibKernel/SceLibKernel.cpp

```diff
@@ -1092,36 +1086,36 @@ EXPORT(int, sceKernelGetTimerTime) {
  * \param error_val Error value on failure
  * \return True on success, false on failure
  */
-bool load_module(SceUID &mod_id, Ptr<const void> &entry_point, SceKernelModuleInfoPtr &module, HostState &host, const char *export_name, char *path, int &error_val) {
+bool load_module(SceUID &mod_id, Ptr<const void> &entry_point, SceKernelModuleInfoPtr &module, HostState &host, const char *export_name, const char *path, int &error_val) {
     const auto &loaded_modules = host.kernel.loaded_modules;
 
-    SceKernelModuleInfoPtrs::const_iterator module_iter = std::find_if(loaded_modules.begin(), loaded_modules.end(), [path](const auto &p) {
+    auto module_iter = std::find_if(loaded_modules.begin(), loaded_modules.end(), [path](const auto &p) {
         return std::string(p.second->path) == path;
     });
 
     if (module_iter == loaded_modules.end()) {
         // module is not loaded, load it here
 
-        SceUID file = open_file(host.io, path, SCE_O_RDONLY, host.pref_path.c_str(), export_name);
+        const auto file = open_file(host.io, path, SCE_O_RDONLY, host.pref_path, export_name);
         if (file < 0) {
             error_val = RET_ERROR(file);
             return false;
         }
-        int size = seek_file(file, 0, SCE_SEEK_END, host.io, export_name);
+        const auto size = seek_file(file, 0, SCE_SEEK_END, host.io, export_name);
         if (size < 0) {
             error_val = RET_ERROR(SCE_ERROR_ERRNO_EINVAL);
             return false;
         }
 
         if (seek_file(file, 0, SCE_SEEK_SET, host.io, export_name) < 0) {
-            error_val = RET_ERROR(size);
+            error_val = RET_ERROR(static_cast<int>(size));
             return false;
         }
 
-        char *data = new char[size];
+        auto *data = new char[static_cast<int>(size) + 1]; // null-terminated char array
```

Comment:
Why cast here?

Reply:
Partially for logging consistency.

Other lines for size were already casted to int for RET_ERROR, so it would make sense to have this line with size as casted to int as well.

Reply:
Okay :)